components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/service/IdentityKeyStoreGeneratorImpl.java

```diff
@@ -0,0 +1,334 @@
+
+        } catch (Exception e) {
+            String msg = "Error while instantiating a keystore";
+            LOG.error(msg, e);
+            throw new KeyStoreManagementException(msg, e);
+        } finally {
+            if (lockAcquired) { // Only release the lock if it was acquired
+                lock.unlock();
+            }
+            tenantLocks.remove(tenantId); // Clean up locks for this tenant if not needed anymore
+        }
+    }
+
+    private boolean isContextKeyStoreExists(String context) {
+
+        String keyStoreName = KeystoreUtils.getKeyStoreFileLocation(tenantDomain +
+                KEY_STORE_CONTEXT_SEPARATOR + context);
+        boolean isKeyStoreExists = false;
+        try {
+            keyStoreManager.getKeyStore(keyStoreName);
+            isKeyStoreExists = true;
+        } catch (Exception e) {
+            String msg = "Error while checking the existence of keystore.  ";
+            LOG.error(msg + e.getMessage());
+        }
+        return isKeyStoreExists;
+    }
+
+    /**
+     * This method is used to generate a random password for the generated keystore
+     *
+     * @return generated password
+     */
+    private String generatePassword() {
+
+        SecureRandom random = new SecureRandom();
+        String randString = new BigInteger(130, random).toString(12);
+        return randString.substring(randString.length() - 10);
+    }
+
+    /**
+     * Persist the keystore in the gov.registry
+     *
+     * @param keyStore created Keystore of the tenant
+     * @throws KeyStoreManagementException Exception when storing the keystore in the registry
+     */
+    private void persistContextKeyStore(KeyStore keyStore, String context)
+            throws KeyStoreManagementException {
+
+        String keyStoreName = generateContextKSNameFromDomainName(context);
+        try {
+            char[] passwordChar = password.toCharArray();
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            keyStore.store(outputStream, passwordChar);
+            outputStream.flush();
+            outputStream.close();
+
+            keyStoreManager.addKeyStore(outputStream.toByteArray(), keyStoreName,
+                    passwordChar, " ", KeystoreUtils.getKeyStoreFileType(tenantDomain), passwordChar);
+        } catch (SecurityException e) {
+            if (e.getMessage() != null && e.getMessage().contains("Key store " + keyStoreName + " already available")) {
+
+                LOG.warn("Key store " + keyStoreName + " is already available, ignoring.");
+            } else {
+
+                String msg = "Error when adding a keyStore";
+                LOG.error(msg, e);
+                throw new KeyStoreManagementException(msg, e);
+            }
+        } catch (Exception e) {
+
+            String msg = "Error when processing keystore/pub. cert to be stored in registry";
+            LOG.error(msg, e);
+            throw new KeyStoreManagementException(msg, e);
```

Comment:
any particular reason to log and throw? if not only throw

Reply:
Saw few other places in the PR as well. Check them as well.

Reply:
addressed in https://github.com/wso2/carbon-identity-framework/pull/6189/commits/6ca0b3d01b63c8f3f8ee122988af923b204f342b